tests/unit/datasets/test_json_graph_format.py

```
@@ -0,0 +1,428 @@
+        """Test duration without P prefix raises error."""
+        with pytest.raises(ValidationError, match="Type 'duration' requires ISO 8601 format"):
+            PropertyValue(t="duration", v="1Y2M")
+
+    def test_point_wrong_type(self):
+        """Test point with non-dict value raises error."""
+        with pytest.raises(ValidationError, match="Type 'point' requires coordinate object"):
+            PropertyValue(t="point", v="not a point")
+
+    def test_point_missing_coordinates(self):
+        """Test point without proper coordinates raises error."""
+        with pytest.raises(ValidationError, match="Type 'point' requires .* or .*"):
```

Ruff (0.14.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)